packages/components/src/Dialog/Dialog.tsx

```diff
@@ -60,79 +37,57 @@ const CloseButton = ({ className }: CloseButtonProps) => {
   );
 };
 
-/**
- * Search for a direct child that can act as title to improve accessibility.
- */
-const addTitleProps = (
-  children: ReactNode,
-  titleProps: HTMLAttributes<HTMLElement>
-) => {
-  const childs = Children.toArray(children);
-
-  const titleIndex = childs.findIndex(
-    child =>
-      isValidElement(child) &&
-      (child.type === Header || child.type === Headline)
-  );
-
-  // No child found that can act as title
-  if (titleIndex < 0) {
-    console.warn(
-      'No child in <Dialog> found that can act as title for accessibility. Please add a <Header> or <Headline> as direct child.'
-    );
-    return children;
-  }
-
-  // If we found a child, add the titleProps to it
-  const titleChild = cloneElement(
-    childs[titleIndex] as ReactElement<any>,
-    titleProps
-  );
-  childs.splice(titleIndex, 1, titleChild);
+// Dialog Headline
+// ---------------
+interface DialogHeadlineProps extends Omit<HeadlineProps, 'slot'> {}
 
-  return childs;
-};
+const DialogHeadline = ({ children }: DialogHeadlineProps) => (
+  <Headline slot="title">{children}</Headline>
+);
 
 // Props
 // ---------------
-export interface DialogChildProps {
-  close: DialogContextProps['close'];
-  titleProps: HTMLAttributes<HTMLElement>;
-}
-
-export interface DialogProps extends AriaDialogProps {
-  children?: ReactNode | ((props: DialogChildProps) => ReactNode);
+export interface DialogProps extends RAC.DialogProps {
   variant?: string;
   size?: string;
   closeButton?: boolean;
+  isNonModal?: boolean;
 }
 
 // Component
 // ---------------
-export const Dialog = ({
-  children,
+const _Dialog = ({
   variant,
   size,
   closeButton,
+  isNonModal,
   ...props
 }: DialogProps) => {
-  const ref = useRef(null);
-  const { close } = useDialogContext();
-  const { dialogProps, titleProps } = useDialog(props, ref);
-
   const classNames = useClassNames({ component: 'Dialog', variant, size });
+  let state = useContext(OverlayTriggerStateContext);
+
+  let children = props.children;
 
+  if (typeof children === 'function') {
+    children = children({
+      close: state?.close || (() => {}),
+    });
+  }
   return (
-    <div className={classNames.container} {...dialogProps}>
-      {closeButton && <CloseButton className={classNames.closeButton} />}
-      {typeof children === 'function'
-        ? children({ close, titleProps })
-        : props['aria-labelledby']
-        ? children
-        : addTitleProps(children, titleProps)}
-    </div>
+    <Dialog
+      {...props}
+      className={cn(classNames.container, 'relative outline-none')}
```

Comment:
should be the other way around so that the `container` can override the defaults :)